app/UI/relay-detail.tsx

```diff
@@ -285,6 +270,82 @@ export class RelayInformationComponent extends Component<Props, State> {
     }
 }
 
+type MemberListProps = {
+    getProfileByPublicKey: func_GetProfileByPublicKey;
+    getSpaceMembersChan: func_GetSpaceMembersChan;
+    emit: emitFunc<SelectConversation | ViewUserDetail>;
+    isRelayed: boolean;
+    getMemberSet: () => Set<string>;
+};
+
+type MemberListState = {
+    members: Set<string> | undefined;
+};
+
+export class MemberList extends Component<MemberListProps, MemberListState> {
+    state: MemberListState = {
+        members: undefined,
+    };
+    membersStream: Channel<SpaceMember[] | Error> | undefined;
+
+    async componentDidMount() {
+        if (this.props.isRelayed) {
+            this.membersStream = this.props.getSpaceMembersChan();
+            for await (const members of this.membersStream) {
+                if (members instanceof Error) {
+                    console.error(members.message, members.cause);
+                } else {
+                    const pubkeys = members.map((event) => event.member);
+                    await setState(this, { members: new Set<string>(pubkeys) });
+                }
+            }
+        } else {
+            await setState(this, { members: this.props.getMemberSet() });
+        }
```

Comment:
The UI component is doing too much work here.
The better way is to let `getMemberSet` return the correct data for corresponding relay implementations.